src/main/java/org/toradocu/extractor/BlockTag.java

```diff
@@ -11,7 +10,7 @@
  * specification (available after the translation of the comment). A specification is the
  * translation of the comment into a Java boolean condition.
  */
-public abstract class BlockTag<S extends Specification> {
+public abstract class BlockTag {
```

Comment:
See the note on `getSpecifications()`.

 I'm not looking at this in the context of all of the code, but I'm wondering what purpose this class hierarchy serves. The classes represent the kinds of tags in the javadoc, but they are really only wrappers for specifications that are constructed from those tags. If you have already constructed the specifications, why do you need the tag?